apps/zbugs/api/index.ts

```diff
@@ -178,22 +191,35 @@ fastify.post<{
   await Promise.all(postCommitTasks.map(task => task()));
 
   reply.send(response);
-});
+}
 
 fastify.post<{
   Querystring: Record<string, string>;
   Body: ReadonlyJSONValue;
-}>('/api/pull', async (request, reply) => {
+}>('/api/pull', getQueriesHandler);
+
+fastify.post<{
+  Querystring: Record<string, string>;
+  Body: ReadonlyJSONValue;
+}>('/api/get-queries', getQueriesHandler);
+
+async function getQueriesHandler(
+  request: FastifyRequest<{
+    Querystring: Record<string, string>;
+    Body: ReadonlyJSONValue;
+  }>,
+  reply: FastifyReply,
+) {
   await withAuth(request, reply, async authData => {
     reply.send(
-      await mapQueryRequest(
-        async (name, args) => ({query: getQuery(authData, name, args)}),
+      await mapGetQueriesRequest(
```

Comment:
The tiniest of polish but what do you think about `handleGetQueriesRequest`? It is a request handler basically.

Reply:
And of course same change for push.